pyemma/coordinates/clustering/interface.py

```diff
@@ -23,30 +23,52 @@ def __init__(self, metric='euclidean'):
         super(AbstractClustering, self).__init__()
         self.metric = metric
         self.clustercenters = None
-        self.dtrajs = []
+        self._dtrajs = []
+
+    @property
+    def dtrajs(self):
+        if (len(self._dtrajs) == 0):  # nothing assigned yet, doing that now
+            return self.assign(stride = 1)
+        else:
+            return self._dtrajs  # returning what we have saved
 
     def _map_array(self, X):
         """get closest index of point in :attr:`clustercenters` to x."""
-        dtraj = np.empty(X.shape[0], np.int64)
+        dtraj = np.empty(X.shape[0], dtype=self.output_type())
```

Comment:
I actually tried with np.int32 (which is in my opinion really 
sufficient), but then an exception is thrown from the underlying method 
that int64 is expected. So this is covered, but I would prefer it to be 
changed to np.int32. It's not crucial at the moment though.

Am 12/04/15 um 14:44 schrieb Martin K. Scherer:
>
> In pyemma/coordinates/clustering/interface.py 
> <https://github.com/markovmodel/PyEMMA/pull/228#discussion_r28203107>:
>
> >
> >      def _map_array(self, X):
> >          """get closest index of point in :attr:`clustercenters` to x."""
> > -        dtraj = np.empty(X.shape[0], np.int64)
> > +        dtraj = np.empty(X.shape[0], dtype=self.output_type())
>
> I'm not sure if the underlying c code does access arbitrary types from 
> output_type. At least an assertion should go here, that currently only 
> int64 is supported.
>
> |if(PyArray_TYPE(np_dtraj)!=NPY_INT64) { PyErr_SetString(PyExc_ValueError, "dtype of \"dtraj\" isn\'t int (64)."); goto error; };
> |
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/markovmodel/PyEMMA/pull/228/files#r28203107>.
>


-- 

----------------------------------------------
Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de
Mail:  Arnimallee 6, 14195 Berlin, Germany
----------------------------------------------